src/scripts/main.js

```diff
@@ -1 +1,67 @@
 'use strict';
+
+const firstPromise = new Promise((resolve, reject) => {
+  const timeoutId = setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0) {
+      clearTimeout(timeoutId);
+      resolve('First promise was resolved');
+    }
+  });
+});
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('mousedown', (ev) => {
+    if (ev.button === 0 || ev.button === 2) {
+      resolve('Second promise was resolved');
+    }
+  });
+});
```

Comment:
The `secondPromise` resolves on either a left or right mouse click. This may cause the promise to resolve multiple times if both buttons are clicked separately. Consider adding a mechanism to ensure the promise resolves only once.